src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveHandler.cs

```diff
@@ -0,0 +1,98 @@
+    [Shared]
+    [ExportLspMethod(LSP.Methods.TextDocumentCompletionResolveName)]
+    internal class CompletionResolveHandler : IRequestHandler<LSP.CompletionItem, LSP.CompletionItem>
+    {
+        public async Task<LSP.CompletionItem> HandleRequestAsync(Solution solution, LSP.CompletionItem completionItem,
+            LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            CompletionResolveData data;
+            if (completionItem.Data is CompletionResolveData)
+            {
+                data = (CompletionResolveData)completionItem.Data;
+            }
+            else
+            {
+                data = ((JToken)completionItem.Data).ToObject<CompletionResolveData>();
+            }
+
+            var request = data.CompletionParams;
+
+            var document = solution.GetDocumentFromURI(request.TextDocument.Uri);
+            if (document == null)
+            {
+                return completionItem;
+            }
+
+            var position = await document.GetPositionFromLinePositionAsync(ProtocolConversions.PositionToLinePosition(request.Position), cancellationToken).ConfigureAwait(false);
+
+            var completionService = document.Project.LanguageServices.GetService<CompletionService>();
+            var list = await completionService.GetCompletionsAsync(document, position, cancellationToken: cancellationToken).ConfigureAwait(false);
+            if (list == null)
+            {
+                return completionItem;
+            }
+
+            var selectedItem = list.Items.FirstOrDefault(i => i.DisplayText == data.DisplayText);
+            if (selectedItem == null)
+            {
+                return completionItem;
+            }
+
+            var description = await completionService.GetDescriptionAsync(document, selectedItem, cancellationToken).ConfigureAwait(false);
+
+            LSP.CompletionItem resolvedCompletionItem;
+            if (clientCapabilities?.HasVisualStudioLspCapability() == true)
+            {
+                resolvedCompletionItem = CloneVSCompletionItem(completionItem);
+                ((LSP.VSCompletionItem)resolvedCompletionItem).Description = new ClassifiedTextElement(description.TaggedParts.Select(
+                    tp => new ClassifiedTextRun(tp.Tag.ToClassificationTypeName(), tp.Text)));
+            }
+            else
+            {
+                resolvedCompletionItem = RoslynCompletionItem.From(completionItem);
+                ((RoslynCompletionItem)resolvedCompletionItem).Description = description.TaggedParts.Select(
+                    tp => new RoslynTaggedText { Tag = tp.Tag, Text = tp.Text }).ToArray();
+            }
+            resolvedCompletionItem.Detail = description.TaggedParts.GetFullText();
+
+            return resolvedCompletionItem;
+        }
+
+        private LSP.VSCompletionItem CloneVSCompletionItem(LSP.CompletionItem completionItem)
+        {
+            return new LSP.VSCompletionItem
```

Comment:
so LSP and LSP.Extension types in same nuget?

Reply:
Same namespace, different nuget